botocore/client.py

```diff
@@ -832,27 +820,34 @@ def _resolve_signature_version(self, service_name, resolved):
         if configured_version is not None:
             return configured_version
 
-        potential_versions = resolved.get('signatureVersions', [])
-        if (
-            self.service_signature_version is not None
-            and self.service_signature_version
-            not in _LEGACY_SIGNATURE_VERSIONS
-        ):
-            # Prefer the service model as most specific
-            # source of truth for new signature versions.
-            potential_versions = [self.service_signature_version]
+        # These have since added the "auth" key to the service model
+        # with "aws.auth#sigv4", but preserve existing behavior from
+        # when we preferred endpoints.json over the service models
+        if service_name in ['s3', 's3-control']:
+            return 's3v4'
 
-        # Pick a signature version from the endpoint metadata if present.
-        if 'signatureVersions' in resolved:
-            if service_name == 's3':
-                return 's3v4'
+        if self.service_signature_version is not None:
+            # Prefer the service model
+            potential_versions = [self.service_signature_version]
+            # importexport is modeled with V2, but we previously allowed
+            # endpoints.json to move it to v4 before switching this method
+            # to prefer the service models
+            if service_name == 'importexport':
+                return 'v4'
```

Comment:
Do we have any other options for this? S3 has it's own special cases throughout the code but adding more random service conditionals is ideally avoided. I'd maybe suggest we add a handler to set this explicitly for the service instead.

Reply:
One note on the approach in https://github.com/boto/botocore/pull/3465/commits/b2da232e0219bfa449aa1639daf98f9c338db3f3:

`client.meta.config.signature_version` for `importexport` will now return `v2` (instead of `v4`) since this doesn't execute the `choose-signer` event.

I removed that assertion from the functional test, but left the assertion that the request ultimately has the V4 header.